connection.go

```diff
@@ -527,6 +527,8 @@ runLoop:
 		default:
 		}
 
+		s.updateMaxDatagramDataSize()
```

Comment:
Can we be smarter than running this once per `runLoop` iteration? This is the hot path.

Reply:
Do you mean this should be moved to the end of `runLoop` body? I wasn't sure under which conditions end of the `runLoop` body is or isn't reachable.

Reply:
Not really. `runLoop` iterates once per (batch of) packets received and sent. Which is a lot on an active QUIC connection.

The available space in the packet however only changes under 2 conditions (I think):
1. When we update the connection ID.
2. When we discover a new MTU.

Reply:
I think there's another reason: encoded packet number length increases, which IIUC grows with RTT and how fast we send packets (as in, packets per unit of time.)

I also just realized I misread and misunderstood your initial comment, oops.

The simplest way I can think of would be just to rate limit `updateMaxDatagramSize`. A more "precise" solution would be to run it in response to the 3 events: connection ID changed, PMTU changed, packet number length changed. Packet number length can change after every packet we send and on every ACK we receive. This reasoning is in fact why I initially went with running `updateMaxDatagramSize` every `runLoop`, as it runs in response to all of these events. In any case, do you think just rate limiting so that `updateMaxDatagramSize` runs once in a while, e.g. every 10 ms, or maybe it should be every N iterations of runLoop, would be good enough solution, or should we instead try to run `updateMaxDatagramSize` in response to 3 events listed above?